src/Maestro/Maestro.DataProviders/ConfigurationIngestion/ConfigurationIngestor.cs

```diff
@@ -0,0 +1,347 @@
+            .Include(ns => ns.Channels)
+            .Include(ns => ns.DefaultChannels)
+            .Include(ns => ns.RepositoryBranches)
+            .Where(ns => ns.Name == configurationNamespace)
+            .FirstOrDefaultAsync();
+
+        if (namespaceEntity is null)
+        {
+            namespaceEntity = new Namespace
+            {
+                Name = configurationNamespace,
+                Subscriptions = [],
+                Channels = [],
+                DefaultChannels = [],
+                RepositoryBranches = [],
+            };
+            _context.Namespaces.Add(namespaceEntity);
+            await _context.SaveChangesAsync();
+        }
+
+        return namespaceEntity;
+    }
+
+    private async Task CreateSubscriptions(
+        IEnumerable<IngestedSubscription> newSubscriptions,
+        Namespace namespaceEntity,
+        Dictionary<string, Channel> existingChannelsByName)
+    {
+        List<Subscription> subscriptionDaos = [.. newSubscriptions
+            .Select(sub => ConfigurationDataHelper.ConvertIngestedSubscriptionToDao(
+                sub,
+                namespaceEntity,
+                existingChannelsByName))];
+
+        await _sqlBarClient.CreateSubscriptionsAsync(subscriptionDaos, false);
+    }
+
+    private async Task UpdateSubscriptions(
+        IEnumerable<IngestedSubscription> updatedSubscriptions,
+        Namespace namespaceEntity,
+        Dictionary<string, Channel> existingChannelsByName)
+    {
+        List<Subscription> subscriptionDaos = [.. updatedSubscriptions
+            .Select(sub => ConfigurationDataHelper.ConvertIngestedSubscriptionToDao(
+                sub,
+                namespaceEntity,
+                existingChannelsByName))];
+
+        await _sqlBarClient.UpdateSubscriptionsAsync(subscriptionDaos, false);
+    }
+
+    private async Task DeleteSubscriptions(IEnumerable<IngestedSubscription> subscriptionRemovals)
+    {
+        var subscriptionIds = subscriptionRemovals.Select(sub => sub.Values.Id).ToList();
+
+        var subscriptionDaos = await _context.Subscriptions
+            .Where(sub => subscriptionIds.Contains(sub.Id))
+            .ToListAsync();
+
+        _context.Subscriptions.RemoveRange(subscriptionDaos);
+    }
+
+    private void CreateChannels(
+        IEnumerable<IngestedChannel> newChannels,
+        Namespace namespaceEntity)
+    {
+        List<Channel> channelDaos = [.. newChannels
+            .Select(ch => ConfigurationDataHelper.ConvertIngestedChannelToDao(ch, namespaceEntity))];
+
+        _context.Channels.AddRange(channelDaos);
+    }
+
+    private void UpdateChannels(
+        IEnumerable<IngestedChannel> updatedChannels,
+        List<Channel> dbChannels,
+        Namespace namespaceEntity)
+    {
+        var dbChannelsByName = dbChannels.ToDictionary(c => c.Name);
+
+        foreach (var channel in updatedChannels)
+        {
+            dbChannelsByName.TryGetValue(channel.Values.Name, out Channel? dbChannel);
```

Comment:
if we know it's there, this could be
```suggestion
            var dbChannel = dbChannelsByName[channel.Values.Name]
```
and we don't need the `!` bellow